### DISCUSSION TITLE
Add a sync plugin for Discogs

### DISCUSSION THREAD

**Discussion by evur** [beetbox/beets#4680] [Ideas]
Currently there is the mbsync plugin for synchronizing metadata from MusicBrainz. But such a plugin is completely missing for Discogs.

The solutions I have in mind are:
1. to replace the mbsync plugin with a plugin which syncs from both sources
or
2. to add a Discogs sync plugin and a third plugin which calls the correct plugin, depending on the <s>availability of the IDs</s> metadata source.

**Comment by sampsyo** [comment]
Interesting idea! Both approaches seem reasonable. It would be interesting to see how much functionality could be factored out and shared between the two sync backends—it's not actually clear to me how well the current code (which is surprisingly complex for what it is) can be disentangled from the MusicBrainz parts. It would be interesting to find out, regardless of which approach we take!

**Comment by evur** [comment]
@sampsyo Thank you for taking interest in the idea!

I think there still is an issue blocking this and everything else related to new plugins interacting with Discogs (like #4676):
in order to access the Discogs API, beets needs to be authenticated. Of course we don't want duplicate code. So I think the Discogs authentication code and config should first be seperated.

**Comment by sindreruud** [comment]
This would be amazing. I almost exclusively use Discogs as the source for my metadata and tagging, so I have really missed an alternative to mbsync that works with Discogs.
